desc/magnetic_fields.py

```diff
@@ -113,6 +175,141 @@ def __call__(self, coords, params={}, basis="rpz"):
+            grid_M and grid_N
+        grid_M : int, optional
+            Poloidal resolution of the grid on the surface
+            on which to calculate Bnormal, by default 40
+        grid_N : int, optional
+            Toroidal resolution of the grid on the surface
+            on which to calculate Bnormal, by default 40
+        NFP : int, optional
+            Number of field periods for compute grid, if None
+            defaults to surface.NFP.
+
+        Returns
+        -------
+        None
+        """
+        if NFP is None:
+            NFP = surface.NFP
+        if grid is None:
+            grid = LinearGrid(rho=jnp.array(1.0), M=grid_M, N=grid_N, NFP=NFP)
+
+        coords = surface.compute_coordinates(grid=grid, basis="xyz")
+        rs_t = surface.compute_coordinates(grid=grid, dt=1, basis="xyz")
+        rs_z = surface.compute_coordinates(grid=grid, dz=1, basis="xyz")
+        surf_normal = cross(rs_t, rs_z)
+        B = self.compute_magnetic_field(coords, basis="xyz")
+
+        Bnorm = jnp.sum(B * surf_normal, axis=-1)
+        return Bnorm
+
+    def save_BNORM_file(
+        self,
+        eq,
+        fname,
+        basis_M=24,
+        basis_N=24,
+        grid=None,
+        sym="sin",
+        scale_by_curpol=True,
+    ):
+        """Create BNORM-style .txt file containing Bnormal Fourier coefficients.
+
+        Parameters
+        ----------
+        eq : Equilibrium
+            Equilibrium to calculate the magnetic field's Bnormal on the surface of.
+        fname : str
+            name of file to save the BNORM Bnormal Fourier coefficients to.
+        basis_M : int, optional
+            Poloidal resolution of the DoubleFourierSeries used to fit the Bnormal
+            on the plasma surface, by default 24
+        basis_N : int, optional
+            Toroidal resolution of the DoubleFourierSeries used to fit the Bnormal
+            on the plasma surface, by default 24
+        grid : Grid, optional
+            Grid of points on the plasma surface to calculate the Bnormal at and then
+            fit with a DoubleFourierSeries, if None defaults to a LinearGrid with twice
+            the basis poloidal and toroidal resolutions
+        sym : str, optional
+            if Bnormal is symmetric, by default "sin"
+            NOTE: BNORM code only ever deals with sin-symmetric modes, so results
+            may not be as expected if attempt to create a BNORM file with a
+            non-symmetric Bnormal distribution, as only the sin-symmetric modes
+            will be saved.
+        scale_by_curpol : bool, optional
+            Whether or not to scale the Bnormal coefficients by curpol
+            which is expected by most other codes that accept BNORM files,
+            by default True
+
+        Returns
+        -------
+        None
+        """
+        if sym != "sin":
+            raise UserWarning(
+                "BNORM code assumes that |B| has sin symmetry,"
```

Comment:
does this mean it wouldn't work for stellarator asymmetric cases? Should we even include the sym option?

Reply:
I am unsure, the BNORM code assumes this so any output that is not stell sym I am not even sure how to represent in their format.